src/core.jl

```diff
@@ -135,23 +136,32 @@ struct RecvBuf
     lcm::Ptr{Cvoid}
 end
 
-function onresponse(rbuf::RecvBuf, channelbytes::Ptr{UInt8}, opts::SubscriptionOptions{T}) where T
-    channel = unsafe_string(channelbytes)
+function check_channel_name(channelptr::Ptr{UInt8}, opts::SubscriptionOptions)
+    i = 1
+    while (byte = unsafe_load(channelptr)) != 0x00
+        byte == UInt8(opts.channel[i]) || error("Mismatch between received channel name and subscription channel name")
```

Comment:
I think `codeunit(opts.channel, i)` would be more appropriate than `UInt8(opts.channel[i])`, as that will return exactly one byte even if the channel is non-ASCII. It's also `O(1)` while indexing into a string is `O(N)`.

Reply:
Great, was looking for something like this but wasn't sure where to look.

Reply:
Oops, I thought you said `codeunits(string, i)` which doesn't exist so I cached the result of `codeunits(string)`. I'll fix it.